tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/BuildersVariables.cs

```diff
@@ -0,0 +1,815 @@
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 66 } }", 768)]
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 67 } }", 769)]
+        [BuildersMQL("{ \"Root.Data\" : 26 }", 771)]
+        [BuildersMQL("{ \"Root.Data\" : 36 }", 771)]
+        [BuildersMQL("{ \"Age\" : { \"$lt\" : 21 } }", 771)]
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 68 } }", 771)]
+        [BuildersMQL("{ \"Root.Data\" : 25 }", 773)]
+        [BuildersMQL("{ \"Root.Left.Data\" : { \"$lte\" : 22 } }", 774)]
+        [BuildersMQL("{ \"Root.Data\" : { \"$ne\" : 20 } }", 776)]
+        [BuildersMQL("{ \"Age\" : { \"$lt\" : 21 } }", 777)]
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 68 } }", 779)]
+        public void Nested_conditional_expressions()
+        {
+            var w = Builders<Tree>.Filter.Eq(t => t.Root.Data, 25);
+            var x = Builders<Tree>.Filter.Lte(t => t.Root.Left.Data, 22);
+            var z = Builders<Tree>.Filter.Ne(t => t.Root.Data, 20);
+            var a = Builders<User>.Filter.Lt(u => u.Age, 21);
+            var b = Builders<User>.Filter.Gt(u => u.Height, 65);
+            var c = Builders<User>.Filter.Gt(u => u.Height, 68);
+            var d = Builders<User>.Filter.Gt(u => u.Height, 66);
+            var e = Builders<User>.Filter.Gt(u => u.Height, 67);
+
+            var y = w == Builders<Tree>.Filter.Eq(t => t.Root.Data, 26) ? z == Builders<Tree>.Filter.Eq(t => t.Root.Data, 36) ? e = a: d = e : b = c;
+
+            var finalResultW = w;
+            var finalResultX = x;
+            var finalResultY = y;
+            var finalResultZ = z;
+            var finalResultA = a;
+            var finalResultB = b;
+            var finalResultC = c;
+            var finalResultD = d;
+            var finalResultE = e;
+        }
+
+        [BuildersMQL("{ \"Root.Left.Data\" : { \"$lte\" : 22 } }", 794)]
+        [BuildersMQL("{ \"Root.Data\" : 25 }", 796)]
+        [BuildersMQL("{ \"Address\" : { \"$exists\" : true } }", 800)]
+        [BuildersMQL("{ \"Address\" : { \"$exists\" : false } }", 801)]
+        [BuildersMQL("{ \"Address\" : { \"$exists\" : true } }", 802)]
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 67 } }", 804)]
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 65 } }", 805)]
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 67 } }", 806)]
+        public void Member_access_and_element_access_should_not_be_displayed()
+        {
+            var x = Builders<Tree>.Filter.Lte(t => t.Root.Left.Data, 22);
+            var array = new FilterDefinition<Tree>[1];
+            array[0] = Builders<Tree>.Filter.Eq(t => t.Root.Data, 25);
+            var arrayRef = array;
+
+            var nestedFilter = new NestedFilter();
+            var PersonFilter = Builders<Person>.Filter.Exists(p => p.Address, true);
+            nestedFilter.PersonFilter = Builders<Person>.Filter.Exists(p => p.Address, false);
+            var finalResultPersonFilter = PersonFilter;
```

Comment:
Maybe put some comment about same variable and member names?

Reply:
Done